oak_idl/src/lib.rs

```diff
@@ -18,21 +18,61 @@
 
 extern crate alloc;
 
-use alloc::vec::Vec;
+use alloc::{string::String, vec::Vec};
 use core::fmt::Debug;
 
 pub mod utils;
 
+// TODO(#2500): Align with gRPC Status.
+// Ref: https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto
+/// Logical error model.
 #[derive(Debug)]
-pub enum Error {
+pub struct Error {
+    pub code: ErrorCode,
+    /// English message that helps developers understand and resolve the error.
+    pub message: Option<String>,
```

Comment:
nit: I personally think having only a string is sufficient as it can be empty. That is unless there is a logical distinction between None and Some(""), in which case it would be worth documenting the difference.

Reply:
Thanks, that's a good point. Let's go with just `String` then.